### PR TITLE
copr: Vectorize InetNtoa

### PR DESCRIPTION
**Pull request by waynexia**
PCP #5751 

Signed-off-by: Ruihang Xia <waynestxia@gmail.com>

###  What have you changed?

Vectorize InetNtoa

###  What is the type of the changes?

Pick one of the following and delete the others:

- Improvement (a change which is an improvement to an existing feature)

###  How is the PR tested?
Please select the tests that you ran to verify your changes:
- Unit test
- Integration test

###  Does this PR affect documentation (docs) or should it be mentioned in the release notes?

No

###  Does this PR affect `tidb-ansible`?

No

### CODE REVIEW

File: components/tidb_query/src/rpn_expr/impl_miscellaneous.rs [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+use crate::Result;
+
+use std::net::Ipv4Addr;
+
+#[rpn_fn]
+#[inline]
+pub fn inet_ntoa(arg: &Option<Int>) -> Result<Option<Bytes>> {
+    Ok(match arg {
+        Some(arg) => {
+            if *arg < 0 || *arg > i64::from(u32::max_value()) {
```
Comment (sticnarf) [inline components/tidb_query/src/rpn_expr/impl_miscellaneous.rs]: You may use `u32::try_from`

File: components/tidb_query/src/rpn_expr/impl_miscellaneous.rs [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+                None
+            } else {
+                let v = *arg as u32;
+                let ipv4_addr =
+                    Ipv4Addr::new((v >> 24) as u8, (v >> 16) as u8, (v >> 8) as u8, v as u8);
```
Comment (sticnarf) [inline components/tidb_query/src/rpn_expr/impl_miscellaneous.rs]: Is it necessary to convert it to `Ipv4Addr` first? Why not construct the output string directly?

File: components/tidb_query/src/rpn_expr/impl_miscellaneous.rs [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+            (Some(167773449), Some(Bytes::from("10.0.5.9"))),
+            (Some(2063728641), Some(Bytes::from("123.2.0.1"))),
+            (Some(0), Some(Bytes::from("0.0.0.0"))),
+            (Some(545460846593), None),
+            (Some(-1), None),
+            (
+                Some(i64::from(u32::max_value())),
+                Some(Bytes::from("255.255.255.255")),
+            ),
+        ];
```
Comment (sticnarf) [inline components/tidb_query/src/rpn_expr/impl_miscellaneous.rs]: What about `None` input case?

File: components/tidb_query/src/rpn_expr/impl_miscellaneous.rs (hunk 2 above)
Comment (sticnarf) [reply components/tidb_query/src/rpn_expr/impl_miscellaneous.rs]: Another option is using `From<u32>` to create `Ipv4Addr` and formating it to string

File: components/tidb_query/src/rpn_expr/impl_miscellaneous.rs [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+        for (arg, expected) in test_cases {
+            let output = RpnFnScalarEvaluator::new()
+                .push_params(arg)
```
Comment (sticnarf) [inline components/tidb_query/src/rpn_expr/impl_miscellaneous.rs]: You should use `push_param`

**Comment by sticnarf** [comment]
By the way, I find the function is not included in the randgen sqls.

**Comment by breeswish** [comment]
@sticnarf Nice catch. I will add it.

**Comment by sticnarf** [comment]
Please fix the conflicts.

File: components/tidb_query/src/rpn_expr/impl_miscellaneous.rs [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+use crate::Result;
+
+use std::convert::TryFrom;
+use std::net::Ipv4Addr;
+
+#[rpn_fn]
+#[inline]
+pub fn inet_ntoa(arg: &Option<Int>) -> Result<Option<Bytes>> {
+    Ok(arg
+        .and_then(|arg| u32::try_from(arg).ok())
```
Comment (breeswish) [inline components/tidb_query/src/rpn_expr/impl_miscellaneous.rs]: I see that your implementation is different for the non-vectorized version. Could you change it as well to keep two implementations identical?

**Comment by breeswish** [comment]
@sticnarf PTAL

**Pull request merged**